ruby/ruby/Ice/ProxyFunctions.rb

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) ZeroC, Inc.
+
+require_relative 'Proxy.rb'
+require_relative 'OperationMode.rb'
+require_relative 'BuiltinSequences.rb'
+
+module ::Ice
+    module ObjectPrx_mixin
+        def ice_ping(context=nil)
+            ObjectPrx_mixin::OP_ice_ping.invoke(self, [], context)
+        end
+
+        def ice_ids(context=nil)
+            return ObjectPrx_mixin::OP_ice_ids.invoke(self, [], context)
+        end
+
+        def ice_isA(id, context=nil)
+            return ObjectPrx_mixin::OP_ice_isA.invoke(self, [id], context)
+        end
+
+        def ice_id(context=nil)
+            ObjectPrx_mixin::OP_ice_id.invoke(self, [], context)
+        end
+    end
+
+    class ObjectPrx
```

Comment:
Why do we also define ObjectPrx here?

Reply:
We reopen the class to add the 4 builtin methods defined in the mixin above.